iron-input.html

```diff
@@ -238,11 +244,12 @@
         return;
 
       // Check the pattern either here or in `_onInput`, but not in both.
-      this._patternAlreadyChecked = true;
+      this._patternAlreadyChecked = false;//true;
 
       var thisChar = String.fromCharCode(event.charCode);
       if (this._isPrintable(event) && !regexp.test(thisChar)) {
         event.preventDefault();
+        this._announceInvalidCharacter('Invalid character ' + thisChar + ' not entered.');
```

Comment:
Oh, now i see, it means "the character x is invalid and was not entered". 
Yeah your version is more concise :+1: